### PR TITLE
Upgrade android embedding

### LINKED ISSUE #30: Conflicting info in readme and pubspec.yaml
Currently the readme states that the library supports Android and iOS. In the pubspec.yaml file, there is a comment that states that iOS is not supported.

I assume the comment in pubspec.yaml is incorrect, but I leave that up to maintainers to decide / correct.

### LINKED ISSUE #37: Flutter 2.5.0 - Android embedding deprecation warning
Hello, thx for pkg !

We upgraded our flutter version to 2.5.0 and now build or pkg get cmds are printing this msg:
```
The plugin `flutter_nordic_dfu` uses a deprecated version of the Android embedding.
To avoid unexpected runtime failures, or future build failures, try to see if this plugin supports the Android V2 embedding. Otherwise, consider removing it since a future release of Flutter will remove these deprecated APIs.
If you are plugin author, take a look at the docs for migrating the plugin to the V2 embedding: https://flutter.dev/go/android-plugin-migration.
```

Do you have time to implement the necessary changes ? For now it should still work properly but may break in the future...

Regards,
R0m4in

### PR DESCRIPTION
**Pull request by TNorbury**
This upgrades the plugin's Android embedding to V2.

I've tested it and everything appears to be hunky-dory.

I also changed the Android package name from `com.timeyaa.flutternordicdfu` to `com.timeyaa.flutter_nordic_dfu` as this lines up with the method channel name.

Also added the Flutter Lints pkg and made the change it requested.

closes #37 
updated pubspec to close #30

### CODE REVIEW

File: android/src/main/java/com/timeyaa/flutter_nordic_dfu/FlutterNordicDfuPlugin.java [hunk 1]
```diff
@@ -114,14 +177,14 @@ public void onMethodCall(MethodCall call, Result result) {
...
 
         DfuServiceInitiator starter = new DfuServiceInitiator(address)
                 .setZip(filePath)
                 .setKeepBond(true)
-                .setForceDfu(forceDfu == null ? false:forceDfu)
-                .setPacketsReceiptNotificationsEnabled(enablePRNs == null ? Build.VERSION.SDK_INT < Build.VERSION_CODES.M:enablePRNs)
-                .setPacketsReceiptNotificationsValue(numberOfPackets== null ? -1 :numberOfPackets)
+                .setForceDfu(forceDfu == null ? false : forceDfu)
+                .setPacketsReceiptNotificationsEnabled(enablePRNs == null ? Build.VERSION.SDK_INT < Build.VERSION_CODES.M : enablePRNs)
+                .setPacketsReceiptNotificationsValue(numberOfPackets == null ? 0 : numberOfPackets)
```
Comment (TNorbury) [inline android/src/main/java/com/timeyaa/flutter_nordic_dfu/FlutterNordicDfuPlugin.java:187]: changed from -1 to 0 as the call signature required a number >= 0

File: android/gradle.properties [hunk 2]
```diff
@@ -1,2 +1,2 @@
 org.gradle.jvmargs=-Xmx1536M
-android.enableR8=true
```
Comment (TNorbury) [inline android/gradle.properties:2]: enableR8 is deprecated, so I removed it

File: example/android/app/build.gradle [hunk 3]
```diff
@@ -52,9 +52,6 @@ android {
             // Signing with the debug keys for now, so `flutter run --release` works.
             signingConfig signingConfigs.debug
             minifyEnabled true
-            useProguard true
```
Comment (TNorbury) [inline example/android/app/build.gradle:55]: Proguard is deprecated, so I removed it

**Comment by kent-williams** [comment]
I'm currently getting two errors while building this.
First
```
Error:
	Overlay manifest:package attribute declared at AndroidManifest.xml:2:5-53 value=(com.timeyaa.flutter_nordic_dfu_example)
	has a different value=(com.timeyaa.flutter_nordic_dfu_example) declared in main manifest at AndroidManifest.xml:2:5-45
	Suggestion: remove the overlay declaration at AndroidManifest.xml 	and place it in the build.gradle:
		flavorName {
			applicationId = "com.timeyaa.flutter_nordic_dfu_example"
		}
```
I resolve this by just removing the package definition in the AndroidManifest.xml

Second
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeDexDebug'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
     Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
     Type com.timeyaa.flutter_nordic_dfu.BuildConfig is defined multiple times: /flutter-nordic-dfu/example/build/flutter_nordic_dfu/.transforms/9ed961ef2d2ff20270f5b49725d8bd8f/classes/classes.dex, /flutter-nordic-dfu/example/build/app/intermediates/project_dex_archive/debug/out/com/timeyaa/flutter_nordic_dfu/BuildConfig.dex
```
I haven't resolved this one yet.

**Comment by TNorbury** [comment]
Ok, I think I know what I did wrong. Gonna test a fix real quick and push it as soon as I verify that it works.

@kent-williams just wanna confirm, you get this when building the example, correct?

**Comment by kent-williams** [comment]
> Ok, I think I know what I did wrong. Gonna test a fix real quick and push it as soon as I verify that it works.
> 
> @kent-williams just wanna confirm, you get this when building the example, correct?

Yes that's correct, apologies should've specified!

**Comment by TNorbury** [comment]
@kent-williams No worries! 
I think I fixed it with that last commit, I was able to build the example on my phone, and verified that I was able to DFU something for good measure

**Comment by kent-williams** [comment]
Those fixes did it! thank you!

**Pull request merged**